R/calculate_go_enrichment.R

```diff
@@ -55,24 +58,49 @@ go_enrichment <- function(...) {
 #' organism (TaxId). Possible inputs include only: "9606" (Human), "559292" (Yeast) and "83333"
 #' (E. coli). Is only necessary if GO data is not provided either by \code{go_annotations_uniprot}
 #' or in \code{go_data}.
-#' @param go_data Optional, a data frame that can be obtained with \code{fetch_go}. If you provide
-#' data not obtained with \code{fetch_go} make sure column names for protein ID (db_id) and GO ID
-#' (go_id) are the same as for data obtained with \code{fetch_go}.
+#' @param go_data Optional, a data frame that can be obtained with `fetch_go()`. If you provide
+#' data not obtained with `fetch_go()` make sure column names for protein ID (`db_id`) and GO ID
+#' (`go_id`) are the same as for data obtained with `fetch_go()`.
 #' @param plot a logical argument indicating whether the result should be plotted or returned as a table.
+#' @param plot_style a character argument that specifies the plot style. Can be either "barplot" (default)
+#' or "heatmap". The "heatmap" plot is especially useful for the comparison of multiple groups. We recommend,
+#' however, that you use it only with `enrichment_type = "enriched"` or `enrichment_type = "deenriched`,
+#' because otherwise it is not possible to distinguish between enrichment and deenrichment in the plot.
+#' @param plot_title a character value that specifies the title of the plot. The default is "Gene ontology
+#' enrichment of significant proteins".
+#' @param barplot_fill_colour a vector that contains two colours that should be used as the fill colours for
+#' deenriched and enriched GO terms, respectively. If `enrichment_type = "enriched"` or `"deenriched`, please
+#' still provide two values in the vector, the colour not used for the plot can be `NA` in this case. E.g.
+#' `c(NA, "red")` for `enrichment_type = "enriched"`.
+#' @param heatmap_fill_colour a vector that contains colours that should be used to create the gradient in the
+#' heatmap plot. Default is `mako_colours`.
+#' @param heatmap_fill_colour_rev a logical value that specifies if the provided colours in `heatmap_fill_colour`
+#' should be reversed in order. Default is `TRUE`.
 #' @param label a logical argument indicating whether labels should be added to the plot.
 #' Default is TRUE.
+#' @param enrichment_type a character argument that is either "all", "enriched" or "deenriched". This
+#' determines if the enrichment analysis should be performed in order to check for both enrichemnt and
+#' deenrichemnt or only one of the two. This affects the statistics performed and therefore also the displayed
+#' plot.
+#' @param min_n_detected_proteins_in_process is a numeric argument that specifies the minimal number
```

Comment:
i would rephrase this argument slightly to make it more clear/easier to read: specifies the minimum number of detected proteins required for a GO term to be displayed in the plot. - or something like this